extensions/amp-pan-zoom/0.1/amp-pan-zoom.js

```diff
@@ -166,11 +166,12 @@ export class AmpPanZoom extends AMP.BaseElement {
         return;
       }
       const scale = args['scale'] || 1;
-      const x = args['x'] || 0;
-      const y = args['y'] || 0;
-      const deltaX = x - this.posX_;
-      const deltaY = y - this.posY_;
-      this.onZoom_(scale, deltaX, deltaY, true);
+      this.updatePanZoomBounds_(scale);
+      const x = this.boundX_(args['x'] || 0);
+      const y = this.boundY_(args['y'] || 0);
+      this.set_(scale, x, y, /*animate*/true).then(() => {
```

Comment:
if you return the promise, it would do automatic action chaning so next action won't run until this is done. it seems like the desired behavior in this case.

Reply:
+1.

Reply:
Fixed that.